ingestion/src/metadata/ingestion/source/messaging/kafka.py

```diff
@@ -100,90 +79,77 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
-                        iter(topic_configResource.values())
-                    ):
-                        config_response = j.result(timeout=10)
-                        if "max.message.bytes" in config_response:
-                            topic.maximumMessageSize = config_response.get(
-                                "max.message.bytes", {}
-                            ).value
-
-                        if "min.insync.replicas" in config_response:
-                            topic.minimumInSyncReplicas = config_response.get(
-                                "min.insync.replicas"
-                            ).value
-
-                        if "retention.ms" in config_response:
-                            topic.retentionTime = config_response.get(
-                                "retention.ms"
-                            ).value
-
-                        if "cleanup.policy" in config_response:
-                            cleanup_policies = config_response.get(
-                                "cleanup.policy"
-                            ).value
-                            topic.cleanupPolicies = cleanup_policies.split(",")
-
-                        topic_config = {}
-                        for key, conf_response in config_response.items():
-                            topic_config[key] = conf_response.value
-                        topic.topicConfig = topic_config
-
-                    if topic_schema is not None:
-                        topic.schemaText = topic_schema.schema_str
-                        if topic_schema.schema_type == "AVRO":
-                            topic.schemaType = SchemaType.Avro.name
-                            if self.generate_sample_data:
-                                pass
-                                topic.sampleData = self._get_sample_data(topic.name)
-                        elif topic_schema.schema_type == "PROTOBUF":
-                            topic.schemaType = SchemaType.Protobuf.name
-                        elif topic_schema.schema_type == "JSON":
-                            topic.schemaType = SchemaType.JSON.name
-                        else:
-                            topic.schemaType = SchemaType.Other.name
-
-                    self.status.topic_scanned(topic.name.__root__)
-                    yield topic
-                else:
-                    self.status.dropped(topic_name)
-            except Exception as err:
-                logger.error(repr(err))
-                logger.debug(traceback.format_exc())
-                logger.debug(sys.exc_info()[2])
-                self.status.failure(topic_name, repr(err))
+            yield KafkaTopicDetails(topic_name, topic_metadata)
+
+    def get_topic_name(self, topic_details: KafkaTopicDetails) -> str:
+        """
+        Get Topic Name
+        """
+        return topic_details.topic_name
+
+    def yield_topic(self, topic_details: KafkaTopicDetails) -> Any:
```

Comment:
we are yielding a Topic here so typing can be changed to Topic

Reply:
@ulixius9 sure i will do that.